client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-upgrade-plan/index.tsx

```diff
@@ -0,0 +1,67 @@
+import { PLAN_BUSINESS, getPlan } from '@automattic/calypso-products';
+import { StepContainer } from '@automattic/onboarding';
+import { UpgradePlan } from 'calypso/blocks/importer/wordpress/upgrade-plan';
+import DocumentHead from 'calypso/components/data/document-head';
+import FormattedHeader from 'calypso/components/formatted-header';
+import { useSite } from 'calypso/landing/stepper/hooks/use-site';
+import { useSiteSlug } from 'calypso/landing/stepper/hooks/use-site-slug';
+import { goToCheckout } from 'calypso/landing/stepper/utils/checkout';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import type { Step } from '../../types';
+
+const SiteMigrationUpgradePlan: Step = function ( { navigation } ) {
+	const siteItem = useSite();
+	const siteSlug = useSiteSlug();
+	const plan = getPlan( PLAN_BUSINESS );
+
+	if ( ! siteItem || ! siteSlug || ! plan ) {
+		return;
+	}
+
+	const stepContent = (
+		<UpgradePlan
+			site={ siteItem }
+			ctaText=""
+			subTitleText=""
+			isBusy={ false }
+			hideTitleAndSubTitle
+			onCtaClick={ () => {
+				goToCheckout( {
+					flowName: 'site-migration',
+					stepName: 'site-migration-upgrade-plan',
+					siteSlug: siteSlug,
+					destination: `/setup/site-migration/site-migration-instructions/?flags=onboarding/new-migration-flow&siteSlug=${ siteSlug }`,
```

Comment:
probably not the best way of enabling the feature flag

Reply:
You can enable the feature flag on your env on the calypso env to enable testing on during the development

Reply:
You could also append it to the destination with something like this:

```
`/setup/site-migration/site-migration-instructions/?siteSlug=${ siteSlug }${config.isEnabled('onboarding/new-migration-flow') ? '&flags=onboarding/new-migration-flow' : ''}`
```

Reply:
you can use the import { addQueryArgs } from 'calypso/lib/url';  to add params to a URL safely